src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -924,6 +924,12 @@ private static List<PropertyInfo> GetAllProperties([DynamicallyAccessedMembers(D
 
             var propertyBindingPoint = new BindingPoint(initialValue: config.GetSection(parameterName).Value, isReadOnly: false);
 
+            BindInstance(
```

Comment:
@halter73 @eerhardt the change looks reasonable to me. The only thing I am seeing is it is possible BindInstance throws exceptions which users may start getting when used to get InvalidOperationException before. I am not sure if this is be a concern or would be acceptable.

Reply:
It seems fine. Test `ExceptionWhenTryingToBindToConstructorWithMissingConfig` covers the exception case

Reply:
@tarekgh - do you have specific exception scenarios in mind? I'm not seeing any.

The only thing I can think of that this may break is default values on the ctor parameter. We should have a test for it, if we don't we should add one.

Reply:
The scenario in my head is before the change we used to throw `throw new InvalidOperationException(SR.Format(SR.Error_ParameterHasNoMatchingConfig, type, parameterName));`. Now with the change, we call `BindInstance` first which is possible to throw a different exception (like when calling `TryConvertValue`). So, it is users will get different exceptions. I am not sure if this is a real concern though.

Reply:
Looking more, I am seeing we always throw `InvalidOperationException`. So, i think we fine then. You can ignore this comment.